lib/sync-DataSetModel.js

```diff
@@ -310,26 +316,31 @@ var self = {
   createDatasetClient: function (dataset_id, query_params, meta_data, cb) {
     self.getDataset(dataset_id, function (err, dataset) {
       if (err) return err;
-      var clientHash = dataset_id + "_" + self.getClientHash(query_params, meta_data);
-      var datasetClient = dataset.clients[clientHash];
-      if (!datasetClient) {
-        datasetClient = {
-          id: clientHash,
-          datasetId: dataset_id,
-          created: new Date().getTime(),
-          lastAccessed: new Date().getTime(),
-          queryParams: query_params,
-          metaData: meta_data,
-          syncRunning: false,
-          syncPending: true,
-          syncActive: true,
-          pendingCallbacks: [],
-          instances: {}
-        };
-        dataset.clients[clientHash] = datasetClient;
-      }
-      syncUtil.doLog(dataset_id, 'verbose', 'createDatasetClient :: ' + util.inspect(datasetClient));
-      return cb(null, datasetClient);
+      //var clientHash = dataset_id + "_" + self.getClientHash(query_params, meta_data);
+      //console.log('getDataset callback - Sync : ' + clientHash);
+
+      self.asyncGetClientHash(query_params, meta_data, function(err, clientHash) {
+        console.log('getDataset callback - Async : ' + dataset_id + '_' + clientHash);
+        var datasetClient = dataset.clients[clientHash];
+        if (!datasetClient) {
+          datasetClient = {
+            id: clientHash,
+            datasetId: dataset_id,
+            created: new Date().getTime(),
+            lastAccessed: new Date().getTime(),
```

Comment:
Could we store the date and use it for both `created` and `lastAccessed` and only do one call to `new Date().getTime()`?